raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -393,17 +422,35 @@ public void initialize(
         logger.info("Reading KRaft snapshot and log as part of the initialization");
         partitionState.updateState();
 
-        VoterSet lastVoterSet = partitionState.lastVoterSet();
-        requestManager = new RequestManager(
-            lastVoterSet.voterIds(),
-            quorumConfig.retryBackoffMs(),
-            quorumConfig.requestTimeoutMs(),
-            random
-        );
+        if (requestManager == null) {
+            // The request manager wasn't created using the bootstrap servers
+            // create it using the voters static configuration
+            List<Node> bootstrapNodes = voterAddresses
+                .entrySet()
+                .stream()
+                .map(entry ->
+                    new Node(
+                        entry.getKey(),
+                        entry.getValue().getHostString(),
+                        entry.getValue().getPort()
+                    )
+                )
+                .collect(Collectors.toList());
+
+            logger.info("Starting request manager with bootstrap servers: {}", bootstrapNodes);
```

Comment:
Perhaps we change this message to something like "Staring request manager with static voter set: {}"

Reply:
Done.